### PR TITLE
Add nodeAffinity and tolerations to contour

### PR DESCRIPTION
**Pull request by surajssd**
**Note**: See changes per commit to know the gradual changes made.

Fixes https://github.com/kinvolk/lokomotive/issues/262

### CODE REVIEW

File: docs/configuration-reference/components/contour.md [hunk 1]
```diff
@@ -48,6 +70,8 @@ Example:
 |------------------|---------------------------------------------------------------------------------------------|:-------:|:--------:|
 | `service_monitor`| Create ServiceMonitor for Prometheus to scrape Contour and Envoy metrics.                   | false   | false    |
 | `ingress_hosts`  | [ExternalDNS component](external-dns.md) creates DNS entries from the values provided.      | ""      | false    |
+| `node_affinity`  | Node Affinity for deploying the operator pod and envoy daemonset.                           | -       | false    |
```
Comment (johananl) [inline docs/configuration-reference/components/contour.md:73]: Could you please fix the capitalization (throughout the PR) according to our style guide?

File: pkg/components/contour/template.go [hunk 2]
```diff
@@ -45,5 +45,276 @@ spec:
   selector:
     app: envoy
   type: LoadBalancer
+`,
+
+	"03-contour.yaml": `
```
Comment (invidian) [inline pkg/components/contour/template.go:50]: Why not create small helm chart and use chart templating?

File: docs/configuration-reference/components/contour.md (hunk 1 above)
Comment (surajssd) [reply docs/configuration-reference/components/contour.md]: changed.

File: pkg/components/contour/template.go (hunk 2 above)
Comment (surajssd) [reply pkg/components/contour/template.go:50]: So I tried moving the contour component helm chart. You can find the work done for it here:
https://github.com/kinvolk/lokomotive/pull/new/surajssd/move-contour-to-helm-chart

The problem with it is that we ship grafana dashboard and alertmanager config which conflicts with the helm's templating when used directly.

```console
$ helm template .
Error: parse error at (contour/templates/grafana-dashboard.yaml:103): function "namespace" not defined
```

So these dashboards should either be moved to separate json files and then refer them from the config files. Or escape the conflicting `{{}}` brackets and keep them where the config is. Some inspiration is needed from other charts.

**Pull request merged**